src/nvim/decoration.c

```diff
@@ -95,7 +95,28 @@ void decor_redraw(buf_T *buf, int row1, int row2, Decoration *decor)
   }
 }
 
-void decor_remove(buf_T *buf, int row, int row2, Decoration *decor)
+void decor_add(buf_T *buf, int row, int row2, Decoration *decor)
```

Comment:
Added a `decor_add()` as the dual of `decor_remove()` and consequently renamed the existing `decor_add()` to `decor_push()`. Hope that makes sense.

Reply:
sure, breaking out that function makes sense. #25826 is going to restructure/rename everything tho, so no need to getting the names exactly right now.